app/controllers/application_controller.rb

```diff
@@ -2,9 +2,115 @@
 
 class ApplicationController < Sinatra::Base
 
-  configure do
-    set :public_folder, 'public'
-    set :views, 'app/views'
-  end
+ 	configure do
+	    set :public_folder, 'public'
+	    set :views, 'app/views'
+	    enable :sessions
+		set :session_secret, "password_security"
+	end
+
+	get "/" do
+		erb :index
+	end
+
+	get "/signup" do
+		erb :signup
+	end
+
+	post "/signup" do
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.